process/block/preprocess/balanceComputation.go

```diff
@@ -0,0 +1,106 @@
+package preprocess
+
+import (
+	"math/big"
+	"sync"
+
+	"github.com/ElrondNetwork/elrond-go/process"
+)
+
+type balanceComputation struct {
+	mapAddressBalance map[string]*big.Int
+	mutAddressBalance sync.RWMutex
+}
+
+// NewBalanceComputation creates a new object which computes the addresses balances
+func NewBalanceComputation() (*balanceComputation, error) {
+	return &balanceComputation{
+		mapAddressBalance: make(map[string]*big.Int),
+	}, nil
+}
+
+// Init method resets all the addresses balances
+func (bc *balanceComputation) Init() {
+	bc.mutAddressBalance.Lock()
+	bc.mapAddressBalance = make(map[string]*big.Int)
+	bc.mutAddressBalance.Unlock()
+}
+
+// SetBalanceToAddress method sets balance to an address
+func (bc *balanceComputation) SetBalanceToAddress(address []byte, value *big.Int) {
+	bc.mutAddressBalance.Lock()
+	bc.mapAddressBalance[string(address)] = big.NewInt(0).Set(value)
+	bc.mutAddressBalance.Unlock()
+}
+
+// GetBalanceOfAddress method gets balance of an address
+func (bc *balanceComputation) GetBalanceOfAddress(address []byte) (*big.Int, error) {
```

Comment:
you could eliminate some of the functions.

Reply:
This method is used now for debug info